### ISSUE TITLE
Not getting anything back in Chrome with L5 on nginx

### ISSUE THREAD

**Issue by pdiveris** [itsgoingd/clockwork#118]
I have followed the instructions and can see the itsoingd directory under vendor, have added the class and facade in L5's config/app.php. I have also published its config, set debug to true and confirmed that JSON files are being created under storage/clockwork. However, I am not getting anything back in chrome or firebug. NB: I do not seem to be getting anything either if I try 

server/__clockwork/1452521012.4776.682063426  - should I? Could it be a matter of routes / nginx rewrite rules? I am attaching a file here for you to have a look. 

Thank you, PD

[1452521012.4776.682063426.json.txt](https://github.com/itsgoingd/clockwork/files/85855/1452521012.4776.682063426.json.txt)

**Comment by itsgoingd** [comment]
Hey,

Please check if your app responses contain the `X-Clockwork` headers, you can do this in the Network tab of Dev Tools. If you are missing the headers make sure you added the Clockwork middleware to your Http Kernel.

**Comment by pdiveris** [comment]
Hi,

It's all a bit strange, the response headers contain  the following:

X-Clockwork-Id:1452528064.1895.1056102114
X-Clockwork-Version:1.10

however, I get nothing back in the plugin. Should I be able to view the JSON file over a url, .e. http://something.com/__clockwork/1452528064.1895.1056102114 or something?

Cheers

**Comment by itsgoingd** [comment]
Yep, that's the url, does it return 404?

If you don't mind digging into the code, the routes are registered in [ClockworkServiceProvider](https://github.com/itsgoingd/clockwork/blob/v1/Clockwork/Support/Laravel/ClockworkServiceProvider.php#L33), you can test if that code is executed with simple dd statement. If it's not executed, `$this->app['clockwork.support']->isEnabled()` on [line 24](https://github.com/itsgoingd/clockwork/blob/v1/Clockwork/Support/Laravel/ClockworkServiceProvider.php#L24) returns false for whatever reason. You can debug that further in [ClockworkSupport](https://github.com/itsgoingd/clockwork/blob/v1/Clockwork/Support/Laravel/ClockworkSupport.php#L119).

**Comment by pdiveris** [comment]
Hm, the plot thickens. This code is executed

`$this->app['router']->get('/__clockwork/{id}', 'Clockwork\Support\Laravel\Controllers\CurrentController@getData')->where('id', '[0-9\.]+');`

however, the URL doesn't resolve! I am using nginx and my rewrites / Laravel related bits in the config file are as follows

`location / {
    try_files $uri $uri/ /index.html;
  }

  # unless the request is for a valid file (image, js, css, etc.), send to bootstrap
  if (!-e $request_filename)
  {
      rewrite ^/(.*)$ /index.php?/$1 last;
      break;
  }`

Are you normally using with nginx? Do you mind checking your rewrites?

**Comment by itsgoingd** [comment]
Do your other application routes work? Try defining similar route in your app to see if it works.

My nginx configuration is pretty similar:
```
location / {
    root "/Sites/admin/Clockwork-Laravel-5-Integration-Test/public";
    index  index.php index.html index.htm;
    try_files $uri $uri/ /index.php?$query_string;
}

location ~ \.php$ {
    fastcgi_pass   127.0.0.1:10049;
    fastcgi_index  index.php;
    fastcgi_param  SCRIPT_FILENAME "/Sites/admin/Clockwork-Laravel-5-Integration-Test/public$fastcgi_script_name";
    include        fastcgi_params;
}
```

**Comment by pdiveris** [comment]
Ok, I think I've got an idea now abut what's going on. The nginx rewrite works fine by which I mean that I can define a route which starts with underscores and it will work. There are quite a few routes declared, mostly Dingo based API ones, then a few specific and then finally a catch all route to a ContentController like this:

```
Route::controller('', 'ContentController');
```

It appears that this route takes over the world and it's that controller which throws a (reasonable) 404 (it's my response to a slug that doesn't exist).

Everything is fine since I added an explicit route just above it:

```
Route::get('/__clockwork/{id}', 'Clockwork\Support\Laravel\Controllers\CurrentController@getData')->where('id', '[0-9\.]+');
```

Thanks, P
